docs/ops/deployment/docker.md

```diff
@@ -319,13 +329,57 @@ as described in [how to run the Flink image](#how-to-run-flink-image).
     ENV VAR_NAME value
     ```
 
+    **Commands for building**:
+
     ```sh
     docker build -t custom_flink_image .
     # optional push to your docker image registry if you have it,
     # e.g. to distribute the custom image to your cluster
     docker push custom_flink_image
     ```
 
+    <a name="extend-with-maven"></a>
+    If you need to **extend the Flink image with a Maven dependency (and its transitive dependencies)**,
+you can use a Maven *pom.xml* file such as:
+
+   *pom.xml*:
+
+    ```xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+      <modelVersion>4.0.0</modelVersion>
+
+      <groupId>org.apache.flink</groupId>
+      <artifactId>docker-dependencies</artifactId>
```

Comment:
I believe this is a very frequent task people will have. Even if they know how to download stuff from Maven, having this template available will make the process faster & smoother, because you can just copy paste it, instead of generating and archetype, finding the maven-dependency-plugin etc.

Reply:
This is also relevant for people outside of docker though; so same story as above?